inbm/dispatcher-agent/dispatcher/workload_orchestration_2.py

```diff
@@ -0,0 +1,296 @@
+        The result json determines if the device orchestration status is changed to online mode or not.
+
+        @param return_json: Dict of device status mode and workload details
+        """
+        if return_json["Enabled"]:
+            self._dispatcher_broker.telemetry(
+                'Switching Device Workload Orchestration status to Online mode.')
+            (return_json, status_code) = self.switch_wo_status("false")
+            if status_code != CSL_CMD_STATUS_CODE:
+                self._dispatcher_broker.telemetry(
+                    'Failure in switching Device Workload Orchestration status to Online mode')
+            elif status_code == CSL_CMD_STATUS_CODE:
+                self._dispatcher_broker.telemetry(
+                    'Switched Device Workload Orchestration status to Online mode')
+
+    def _get_workload_orchestration_file_content(self, file: str) -> str:
+        """Reads and returns the workload orchestration file contents
+
+        @return: contents of the file
+        """
+        try:
+            with open(file) as file_content:
+                return file_content.read()
+        except OSError as e:
+            raise DispatcherException(
+                f"Could not load workload orchestration config file with error: {e}")
+
+    def get_wo_details(self) -> Tuple[Optional[str], Optional[str]]:
+        """Returns workload orchestration details
+
+        @return: device workload orchestration ip and token values
+        """
+        token = self.get_orchestrator_value(TOKEN)
+        ip_port = self.get_orchestrator_value(IP)
+        return token, ip_port
+
+    def poll_wo_status(self) -> Tuple[Dict, int]:
+        """Create and process the REST API call to poll device workload orchestration status
+
+        @return: API call return status code and json structure 
+        """
+        (token, ip_port) = self.get_wo_details()
+
+        if token is None or ip_port is None:
+            raise DispatcherException(" Workload-Orchestration IP and Token details Not Found")
+        try:
+            url = self._get_workload_orchestration_file_content(ip_port) + '/api/v1/nodes/' \
+                + WorkloadOrchestration.get_hostname() \
+                + '/maintenance?token=' + self._get_workload_orchestration_file_content(token)
+
+            result = requests.get(url, verify=self.get_orchestrator_value(CSL_CA))
+            return result.json(), result.status_code
+        except (TypeError, AttributeError, OSError, ValueError, JSONDecodeError) as e:
+            raise DispatcherException(str(e))
+        except requests.exceptions.ConnectionError:
+            raise DispatcherException(" Request Connection Error")
+
+    def switch_wo_status(self, maintenance_mode: str) -> Tuple[Dict, int]:
+        """Create and process the Rest API call to switch device into maintenance/online mode
+
```

Comment:
<h3>$\color{red}Defects: Security Vulnerability$</h3>

To improve the security posture, validate the inputs for the `switch_wo_status` method, especially the `maintenance_mode` parameter. Ensuring that it only accepts expected values ('true' or 'false') can prevent potential injection attacks or unintended behavior.